common/persistence/persistenceInterface.go

```diff
@@ -244,6 +246,7 @@ type (
 		SignalInfos         map[int64]*SignalInfo
 		SignalRequestedIDs  map[string]struct{}
 		BufferedEvents      []*DataBlob
+		Checksum            checksum.Checksum
 	}
```

Comment:
i would put checksum in a seperate group, e.g.

```go
InternalWorkflowMutableState struct {
		ExecutionInfo    *InternalWorkflowExecutionInfo
		ReplicationState *ReplicationState
		VersionHistories *DataBlob

		ActivityInfos       map[int64]*InternalActivityInfo
		TimerInfos          map[string]*TimerInfo
		ChildExecutionInfos map[int64]*InternalChildExecutionInfo
		RequestCancelInfos  map[int64]*RequestCancelInfo
		SignalInfos         map[int64]*SignalInfo
		SignalRequestedIDs  map[string]struct{}
		BufferedEvents      []*DataBlob

		Checksum            checksum.Checksum
}
```

Reply:
I assume this is simply personal preference. In general, this is something that is not auto enforcable by gofmt etc. And so, over time, your personal preference for grouping this separately is not sustainable. I will group this separately for now. But philosophically, I am not a big fan these type of semi-enforced groupings (unless there is strong reasons like struct packing in C/C++)